Formula/python@3.10.rb

```diff
@@ -35,12 +35,17 @@ class PythonAT310 < Formula
 
   uses_from_macos "bzip2"
   uses_from_macos "expat"
-  uses_from_macos "libffi"
   uses_from_macos "libxcrypt"
   uses_from_macos "ncurses"
   uses_from_macos "unzip"
   uses_from_macos "zlib"
 
+  on_macos do
+    # libffi added to MacOS SDK on 10.15
+    depends_on "libffi" if MacOS.version < :catalina
+    uses_from_macos "libffi" if MacOS.version >= :catalina
+  end
+
   skip_clean "bin/pip3", "bin/pip-3.4", "bin/pip-3.5", "bin/pip-3.6", "bin/pip-3.7", "bin/pip-3.8", "bin/pip-3.9"
```

Comment:
```suggestion
  skip_clean "bin/pip3", "bin/pip-3.4", "bin/pip-3.5", "bin/pip-3.6", "bin/pip-3.7", "bin/pip-3.8", "bin/pip-3.9"
```
This isn't needed. Also, this changes the dependencies on Linux, which we don't want to do.

Reply:
> This isn't needed. Also, this changes the dependencies on Linux, which we don't want to do.

The `depends_on "libffi" if MacOS.version < :catalina` is required to build on 10.14 and lower SDK as far as I can tell (based on my testing). Also, not sure how it changes dependencies on Linux. Libffi is not being depended on on Linux before or after (can't vouch for if that is correct).

Reply:
> The `depends_on "libffi" if MacOS.version < :catalina` is required to build on 10.14 and lower SDK as far as I can tell (based on my testing).

Sure. That's why you should do
```ruby
uses_from_macos "libffi", since: :catalina
```

> Also, not sure how it changes dependencies on Linux. Libffi is not being depended on on Linux before or after (can't vouch for if that is correct).

```ruby
uses_from_macos "libffi"
```
is (syntatically) equivalent to
```ruby
on_linux do
  depends_on "libffi"
end
```

Said differently, `uses_from_macos` does absolutely nothing on macOS and adds a dependency on Linux.

Reply:
OK, thanks for the pointer. Makes sense now!